connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -527,7 +532,9 @@ public Void call() throws Exception {
 
                         // Note that we use the updated connector config despite the fact that we don't have an updated
                         // snapshot yet. The existing task info should still be accurate.
-                        ConnectorInfo info = new ConnectorInfo(connName, config, configState.tasks(connName));
+                        Connector connector = getConnector(infos.name());
+                        ConnectorInfo info = new ConnectorInfo(connName, config, configState.tasks(connName),
+                            connector instanceof SourceConnector ? ConnectorType.SOURCE : ConnectorType.SINK);
```

Comment:
Why not use `worker.isSinkConnector` here? And if that makes sense, then would it be worthwhile to add a method to the worker that gets the ConnectorType for a connector name?

Reply:
instanceof check is very efficient.
It seems there is no need to involve worker here.

Reply:
Except that it would allow us to centralize the logic and simplify the code. That seems beneficial.